res/enkf/export/summary_collector.py

```diff
@@ -50,32 +49,16 @@ def loadAllSummaryData(ert: EnKFMain, case_name, keys=None, realization_index=No
                 key for key in keys if key in summary_keys
             ]  # ignore keys that doesn't exist
 
-        summary_array = numpy.empty(
-            shape=(len(summary_keys), len(realizations) * len(dates)),
-            dtype=numpy.float64,
-        )
-        summary_array.fill(numpy.nan)
+        ensemble_config = ert.ensembleConfig()
 
-        for key_index, key in enumerate(summary_keys):
-            ensemble_config_node = ert.ensembleConfig().getNode(key)
-            ensemble_data = EnsemblePlotData(ensemble_config_node, fs)
-            summary_row = summary_array[key_index]
-
-            for realization_index, realization_number in enumerate(realizations):
-                realization_vector = ensemble_data[realization_number]
-                column_index = realization_index * len(dates)
-
-                for index in range(1, len(realization_vector)):
-                    if realization_vector.isActive(index):
-                        # assert time_map[index] == realization_vector.getTime(index)
-                        # assert time_map[index].datetime() == dates[index - 1]
-                        value = realization_vector.getValue(index)
-                        summary_row[column_index + index - 1] = value
+        data_array = _lib.enkf_fs_summary_data.get_summary_data(
+            ensemble_config, fs, summary_keys, realizations, len(dates)
+        )
 
         multi_index = MultiIndex.from_product(
             [realizations, dates], names=["Realization", "Date"]
         )
         summary_data = DataFrame(
-            data=numpy.transpose(summary_array), index=multi_index, columns=summary_keys
+            data=numpy.transpose(data_array), index=multi_index, columns=summary_keys
```

Comment:
If you change the ordering of population of the data array in C++ then the transpose is no longer needed